src/types/runtime/gravity/FeaturedLink.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  Record,
+  String,
+  Static,
+  Null,
+  Boolean,
+  Number,
+  Array,
+  Undefined,
+} from "runtypes"
+
+export const FeaturedLink = Record({
+  id: String,
+  _id: String,
+  href: String,
+  title: String,
+  subtitle: String,
+  original_width: Number.Or(Null),
+  original_height: Number.Or(Null),
+  image_url: String.Or(Null),
+  image_versions: Array(String),
+  image_urls: Record({
+    small_rectangle: String.Or(Undefined),
+    large_square: String.Or(Undefined),
+    large_rectangle: String.Or(Undefined),
+    medium_square: String.Or(Undefined),
+    medium_rectangle: String.Or(Undefined),
+    small_square: String.Or(Undefined),
+    wide: String.Or(Undefined),
+  }),
+  display_on_desktop: Boolean,
+  display_on_mobile: Boolean,
+  display_on_martsy: Boolean,
+  created_at: String,
+  updated_at: String,
+})
+
+export type FeaturedLink = Static<typeof FeaturedLink>
```

Comment:
How are these validation schemas created?

Reply:
By just randomly walking through API responses and continuously validating them: https://github.com/artsy/quartz/blob/master/test/stress.ts

Pretty dumb for now, but I have some ideas on how to make it smarter. I haven't moved any of that stuff over yet.